### PR TITLE
Exclude features from IAM preview 

### PR DESCRIPTION
**Pull request by Jeasmine**
* Redisplay, tags, outcomes, impression should not be included on IAM preview

---
This change is [Reviewable](https://reviewable.io/reviews/onesignal/onesignal-ios-sdk/637)

### CODE REVIEW

File: iOS_SDK/OneSignalSDK/Source/OSMessagingController.m [hunk 1]
```diff
@@ -548,11 +574,10 @@ - (BOOL)isClickAvailable:(OSInAppMessage *)message withClickId:(NSString *)click
 
 - (void)sendClickRESTCall:(OSInAppMessage *)message withAction:(OSInAppMessageAction *)action {
     let clickId = action.clickId;
-    // Make sure no click tracking is performed for IAM previews
     // If the IAM clickId exists within the cached clickedClickIds return early so the click is not tracked
     // unless that click is from an IAM with redisplay
     // Handles body, button, or image clicks
-    if (message.isPreview || ![self isClickAvailable:message withClickId:clickId])
+    if (![self isClickAvailable:message withClickId:clickId])
```
Comment (mikechoch) [inline iOS_SDK/OneSignalSDK/Source/OSMessagingController.m]: We are checking for this earlier now right? Thats why we can remove the preview check?

File: iOS_SDK/OneSignalSDK/Source/OSMessagingController.m [hunk 2]
```diff
@@ -531,11 +533,35 @@ - (void)messageViewDidSelectAction:(OSInAppMessage *)message withAction:(OSInApp
...
+    // Make sure no click, outcome, tag tracking is performed for IAM previews
     [self sendClickRESTCall:message withAction:action];
     [self sendTagCallWithAction:action];
     [self sendOutcomes:action.outcomes];
 }
 
+/*
+* Show the developer what will happen with a non IAM preview
+ */
+- (void)logFeaturesNonPreviewIAM:(OSInAppMessage *)message withAction:(OSInAppMessageAction *)action {
```
Comment (mikechoch) [inline iOS_SDK/OneSignalSDK/Source/OSMessagingController.m]: yay this is perfect! lest rename it though from `logFeaturesNonPreviewIAM` to 
`processPreviewIam` or
`processPreviewInAppMessage`

File: iOS_SDK/OneSignalSDK/Source/OSMessagingController.m (hunk 1 above)
Comment (Jeasmine) [reply iOS_SDK/OneSignalSDK/Source/OSMessagingController.m]: thats right

File: iOS_SDK/OneSignalSDK/Source/OSMessagingController.m [hunk 3]
```diff
@@ -531,11 +533,35 @@ - (void)messageViewDidSelectAction:(OSInAppMessage *)message withAction:(OSInApp
...
     [self sendOutcomes:action.outcomes];
 }
 
+/*
+* Show the developer what will happen with a non IAM preview
+ */
+- (void)processPreviewInAppMessage:(OSInAppMessage *)message withAction:(OSInAppMessageAction *)action {
+     if (action.tags) {
+        OSInAppMessageTag *tag = action.tags;
+        if (tag.tagsToAdd)
```
Comment (mikechoch) [inline iOS_SDK/OneSignalSDK/Source/OSMessagingController.m]: Log all tags at once, no need to differentiate or add extra code. Lets do `tag.description` or `tag.jsonRepresentation` so we can just print a string of tags and they can look at the payload.

I have a similar comment on Android PR for this work

**Pull request merged**